plugin/cypress/e2e/navigation.ts

```diff
@@ -14,24 +13,62 @@ When('user clicks on the Service Mesh icon in the left navigation bar', () => {
     })
 });
 
+When('cypress intercept hooks are registred', () => {
+    cy.intercept('/api/proxy/plugin/ossmconsole/kiali/api/namespaces/istio-system/metrics?*').as('metricsRequest')
+    cy.intercept('/api/proxy/plugin/ossmconsole/kiali/api/istio/status?*').as('overviewRequest')
+    cy.intercept('/api/proxy/plugin/ossmconsole/kiali/api/namespaces').as('istioConfigRequest')
+});
+
 Then('buttons for Overview, Graph and Istio Config are displayed', () => {
-    // cy.waitForReact(); // not sure why this is not working
+    cy.waitForReact(5000, '#app', 'node_modules/resq/dist/index.js'); // Manually passing in the resq module path
+    cy.reload(true) // force reload to make sure OSSMC is loaded 
     cy.get('a[data-test="nav"].pf-c-nav__link').contains('Overview')
     cy.get('a[data-test="nav"].pf-c-nav__link').contains('Graph')
     cy.get('a[data-test="nav"].pf-c-nav__link').contains('Istio Config')
-})
+});
 
-Then('user is redirected to the OSSMC {string} page', (hrefName:string) => {
+Then('user is redirected to the OSSMC {string} page', (hrefName: string) => {
     switch (hrefName) {
         case 'Overview':
             cy.get('a[href*="/ossmconsole/overview"]').click().then(() => {
                 cy.url().should('include', "/ossmconsole/overview")
             })
+            break;
         case 'Graph':
             cy.get('a[href*="/ossmconsole/graph"]').click().then(() => {
                 cy.url().should('include', "/ossmconsole/graph")
             })
+            break;
         case 'Istio Config':
-            cy.get('a[href*="/k8s/all-namespaces/istio"]').click().then(() => {})
+            cy.get('a[href*="/k8s/all-namespaces/istio"]').click().then(() => {
+                cy.url().should('include', "/k8s/all-namespaces/istio")
+            })
+            break;
     }
-})
\ No newline at end of file
+});
+
+Then('user sees memory and cpu charts from Kiali', () => {
+    cy.wait('@metricsRequest').then(() => {
+        cy.get('[data-test="memory-chart"]').should('be.visible')
+        cy.get('[data-test="cpu-chart"]').should('be.visible')
+    })
+});
+
+Then('kiali plugin is loaded', () => {
+    cy.wait('@overviewRequest').then(() => {
+        cy.get('h5').contains('istio-system')
```

Comment:
```suggestion
        cy.get('h5').contains('istio-system').should('be.visible')
```

I would add a check statement to make the test fail if the istio-system card is not loaded